lib/griffin/interceptors/server/sentry_interceptor.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+gem "sentry-ruby"
+
+module Griffin
+  module Interceptors
+    module Server
+      class SentryInterceptor < GRPC::ServerInterceptor
+        TRANSACTION_OP = "griffin"
+
+        def request_response(request: nil, call: nil, method: nil)
+          return yield unless Sentry.initialized?
+
+          Sentry.clone_hub_to_current_thread
+
+          Sentry.with_scope do |scope|
+            Sentry.with_session_tracking do
+              scope.clear_breadcrumbs
+
+              service_name = call.service_name
+              method_name = method.name
+              scope.set_transaction_name("#{service_name}/#{method_name}")
+
+              transaction = start_transaction(call.metadata, scope)
+              scope.set_span(transaction) if transaction
+
+              if call.metadata["x-request-id"]
+                scope.set_tags(request_id: call.metadata["x-request-id"])
+              end
+
+              begin
+                response = yield
+              rescue => e
+                if e.is_a?(GRPC::BadStatus)
+                  finish_transaction(transaction, e.code)
+                  raise e
+                end
+
```

Comment:
```suggestion
              rescue GRPC::BadStatus => e
                finish_transaction(transaction, e.code)
                raise e
              rescue => e
```

I think it's better but it's just cosmetic.